llvm/test/CodeGen/NVPTX/store-undef.ll

```diff
@@ -0,0 +1,93 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 5
```

Comment:
Up to you, but personally I feel like, we might as well write the tests manually instead of using this tool?  Or, use the tool, and then delete the lines in the tests that are not relevant?

Like we don't care about the CHECK-EMPTY's and the lines containing comments.

The advantage of this is that, if we ever change the codegen format (and it does change sometimes!), we won't have to regenerate this test and then figure out if the changes are "meaningful".

Reply:
Heh, okay, I defer to Art.  :)

Reply:
Auto-generator is not perfect, but it has two major benefits:
* it's automatic, so the author only need to write the input code, not the checks
* it includes checks for everything, so when something goes wrong I can reason about it without guessing whether the changed output is OK or not, based on hand-written checks that tend to be spotty.

Human-written tests can be good, but usually aren't. The effort of writing a good test falls somewhere between tedious and infeasible. I'll take a good enough automatic test, even if it's overly verbose. It also lowers the review burden as we no longer need to argue about the quality of the checks in the test. Just concentrate on what we're testing and that the generated code is valid.